code/modules/food_and_drinks/restaurant/customers/_customer.dm

```diff
@@ -155,6 +170,108 @@
 	/datum/venue/restaurant = list(/obj/item/food/tofu = 5, /obj/item/food/soup/milo = 6, /obj/item/food/soup/vegetable = 4, /obj/item/food/sashimi = 4, /obj/item/food/chawanmushi = 4, /obj/item/food/meatbun = 4, /obj/item/food/beef_stroganoff = 2),
 	/datum/venue/bar = list(/datum/reagent/consumable/ethanol/beer = 14, /datum/reagent/consumable/ethanol/sake = 9, /datum/reagent/consumable/cafe_latte = 3, /datum/reagent/consumable/coffee = 3, /datum/reagent/consumable/soy_latte = 3, /datum/reagent/consumable/ethanol/atomicbomb = 1))
 
+/datum/customer_data/moth
+	nationality = "Mothman"
+	prefix_file = "strings/names/moth_prefix.txt"
+	base_icon = "mothbot"
+	found_seat_lines = list("Give me your hat!", "Moth?", "Certainly an... interesting venue.")
+	cant_find_seat_lines = list("If I can't find a seat, I'm flippity flapping out of here quick!", "I'm trying to flutter here!")
+	leave_mad_lines = list("I'm telling all my moth friends to never come here!", "Zero star rating, even worse than that time I ate a mothball!","Closing down permanently would still be too good of a fate for this place.")
+	leave_happy_lines = list("I'd tip you my hat, but I ate it!", "I hope that wasn't a collectible!", "That was the greatest thing I ever ate, even better than Guanaco!")
+	wait_for_food_lines = list("How hard is it to get food here? You're even wearing food yourself!", "My fuzzy robotic tummy is rumbling!", "I don't like waiting!")
+	friendly_pull_line = "Moff?"
+	first_warning_line = "Go away, I'm trying to get some hats here!"
+	second_warning_line = "Last warning! I'll destroy you!"
+	self_defense_line = "Flap attack!"
+
+	speech_sound = 'sound/creatures/tourist/tourist_talk_moth.ogg'
+
+	// Always asks for the clothes that you have on, but this is a fallback.
+	orderable_objects = list(
+		/datum/venue/restaurant = list(
+			/obj/item/clothing/head/chefhat = 3,
+			/obj/item/clothing/shoes/sneakers/black = 3,
+			/obj/item/clothing/gloves/color/black = 1,
+		),
+	)
+
+	clothing_sets = list("mothbot_clothes")
+
+	/// The wings chosen for the moth customers.
+	var/list/wings_chosen
+
+// The whole gag is taking off your hat and giving it to the customer.
+// If it takes any more effort, it loses a bit of the comedy.
+// Therefore, only show up if it's reasonable for that gag to happen.
+/datum/customer_data/moth/can_use(datum/venue/venue)
+	return !isnull(get_dynamic_order(venue))
+
+/datum/customer_data/moth/chosen(datum/venue/venue)
+	. = ..()
+
+	// Only show up once, again for the purposes of keeping the comedic value.
+	// Also prevents the unlikely, but otherwise possible, scenario that you keep
+	// getting the extremely easy to satisy moth bots, which isn't as fun as completing real orders.
+	venue.customer_types[type] = 0
```

Comment:
Seeing as tons of other people wanted to add meme characters, make this an option on the base class as a bool.

Reply:
also is there added benefit to keeping them in the list?

Reply:
No, just forgot how to clear keys at the time.